lib/schnorr.py

```diff
@@ -0,0 +1,101 @@
+import os
+import sys
+import traceback
+import ctypes
+from ctypes.util import find_library
+from ctypes import (
+    byref, c_byte, c_int, c_uint, c_char_p, c_size_t, c_void_p, create_string_buffer, CFUNCTYPE, POINTER
+)
+ 
+
+from .util import print_stderr, print_error, print_msg
+
+
+SECP256K1_FLAGS_TYPE_MASK = ((1 << 8) - 1)
+SECP256K1_FLAGS_TYPE_CONTEXT = (1 << 0)
+SECP256K1_FLAGS_TYPE_COMPRESSION = (1 << 1)
+# /** The higher bits contain the actual data. Do not use directly. */
+SECP256K1_FLAGS_BIT_CONTEXT_VERIFY = (1 << 8)
+SECP256K1_FLAGS_BIT_CONTEXT_SIGN = (1 << 9)
+SECP256K1_FLAGS_BIT_COMPRESSION = (1 << 8)
+
+# /** Flags to pass to secp256k1_context_create. */
+SECP256K1_CONTEXT_VERIFY = (SECP256K1_FLAGS_TYPE_CONTEXT | SECP256K1_FLAGS_BIT_CONTEXT_VERIFY)
+SECP256K1_CONTEXT_SIGN = (SECP256K1_FLAGS_TYPE_CONTEXT | SECP256K1_FLAGS_BIT_CONTEXT_SIGN)
+SECP256K1_CONTEXT_NONE = (SECP256K1_FLAGS_TYPE_CONTEXT)
+
+SECP256K1_EC_COMPRESSED = (SECP256K1_FLAGS_TYPE_COMPRESSION | SECP256K1_FLAGS_BIT_COMPRESSION)
+SECP256K1_EC_UNCOMPRESSED = (SECP256K1_FLAGS_TYPE_COMPRESSION)
+
+
+def load_library():
```

Comment:
Do we not already use libsecp for signing?  Let's move this whole thing to a libsecp python module that returns a wrapper class.  In fact, it might make sense to make this a pip module for other people to use if it's possible.

Reply:
Yes, this is not a bad suggestion.  `ecc_fast.py` uses it -- it could be refactored out into `ecc_fast.py` which just does the monkey patching and `secp256k1.py` which finds the library and allocates  a single app-global context.  This is probably the way to go.

As far as pip modules go: there already is ElectrumSV's one but they use core's secp.

I am weaksauce at creating pip modules -- doubly so if they are binary modules -- but ultimately that might not be a bad thing to have done... assuming someone that is pip pypi master can do it.